### DISCUSSION TITLE
How to flash directly to nvme

### DISCUSSION THREAD

**Discussion by gpinheiro-thegoodpenguin** [OE4T/meta-tegra#957] [Q&A]
Is there a known way with meta-tegra to replicate the nvidia flashing nvme method?
[](https://docs.nvidia.com/jetson/l4t/Tegra%20Linux%20Driver%20Package%20Development%20Guide/flashing.html#wwpID0E0OH0HA)

I've managed to flash it using emmc but the image we want to flash is too big for emmc and we want to flash directly to nvme.
I am using the latest dunfell branch.

**Comment by madisongh** [comment]
It should be possible to do something similar to NVIDIA's kernel_flash mechanism by building an image that runs from an initramfs, formats the NVME drive, and enables USB device mode to serve the NVME partitions to the host for writing the contents.

But you'd have to put this together yourself... we don't have handy tools/recipes to make it easy (such a contribution would be welcome!).
